crypto/x509/v3_purp.c

```diff
@@ -83,7 +83,7 @@ int X509_check_purpose(X509 *x, int id, int ca)
     int idx;
     const X509_PURPOSE *pt;
 
-    if (!X509v3_cache_extensions(x, NULL, NULL))
+    if (!X509v3_cache_extensions(x, x->libctx, x->propq))
```

Comment:
I wonder whether we should now remove the `X509v3_cache_extensions` function from the public API. It was recently added. Previously it was an internal function without the libctx/propq parameters...but now we have the ability to store the libctx in the X509 itself, so it seems pointless to pass through a different one.

Reply:
Removed